### PR TITLE
docket: fix URL encoding handling

### PR DESCRIPTION
**Pull request by liam-fitzgerald**
Handles URL encoding special characters on the docket upload side, and when retrieving files from the glob.

cc: @arthyn @Fang- 

Expedited review/deploy would be helpful, this is blocking third party integrations

### CODE REVIEW

File: desk/app/docket.hoon [hunk 1]
```diff
@@ -656,28 +656,40 @@
...
+    =/  fil  (trip fil)
+    =^  ext  fil  (split-at (flop fil) '.')
+    =/  ext  (crip (flop ext))
+    =.  fil  (flop fil)
+    =-  (snoc - ext)
+    %+  turn
+      (scan fil (most fas (star ;~(less fas next))))
+    |=  t=^tape
+    %-  crip
+    (en-urlt:html t)
```
[the remark this answers is not in the document]
Comment (liam-fitzgerald) [reply desk/app/docket.hoon:691]: yeah, sorry lmao, I wrote this under fucked up time pressure at Nearcon

**Comment by arthyn** [comment]
@liam-fitzgerald is it possible that something coming through a clay glob would need to be URL-encoded? if it is possible it seems like this might end up trying to access a file at the wrong name. If it's not possible then this is approved.

**Comment by liam-fitzgerald** [comment]
Nope, see https://github.com/urbit/vere/blob/629ac3698383be5c93324b5ed6355f4aac87dad9/pkg/vere/io/unix.c#L1080

i.e. any file name not parseable by ` (star ;~(pose nud low hep dot sig cab))` never gets picked up by the runtime driver. Applying URLencoding to a filename that will get picked up by the runtime driver will be a no-op.

**Comment by liam-fitzgerald** [comment]
cc: @joemfb

**Comment by joemfb** [comment]
That's right, but that validation line was added after the last network breach -- there is historical data on some ships that violates it. And it can be bypassed from within arvo by using %clay's API directly ... But those are all violations of the hoon types and shouldn't be supported IMO

**Comment by liam-fitzgerald** [comment]
Right, and it's basically unheard of to do anything other than boot a
fakeship, copy data into a desk and glob it, ergo this case is not worth
supporting.

**Review by arthyn — APPROVED** [review]
lgtm!

**Comment by Fang-** [comment]
This breaks clay-built globs with `@`s in their URL.

**Pull request merged**